internal/handlers/common/projection.go

```diff
@@ -234,3 +258,140 @@ func filterFieldArrayElemMatch(k1 string, doc, conditions *types.Document, docVa
 	}
 	return
 }
+
+// filterFieldArraySlice is a function that implements $slice projection query.
+func filterFieldArraySlice(docValue *types.Array, projectionValue any) (*types.Array, error) {
+	switch projectionValue.(type) {
+	// TODO what do we do with int64?
```

Comment:
You could add a test for the `int64` case.

Reply:
An issue with `int64` is that `types.Array.Get(i int)` only operate with `int` 
WDID?

Reply:
done